ublox-bluetooth-commands-mode.go

```diff
@@ -58,13 +59,27 @@ func (ub *UbloxBluetooth) ResetUblox() error {
 // ResetUblox calls the Serial port's ResetViaDTR and does not return until
 // the ublox module has indicated it is ready
 func (ub *UbloxBluetooth) ResetUbloxSync() error {
+	ub.rebootExpected = true
 	err := ub.serialPort.ResetViaDTR()
 
 	if err != nil {
 		return errors.Wrap(err, "[ResetUblox] error")
 	}
 
-	_, err = ub.WaitForResponse(rebootResponseString, false)
+	select {
+	case <-ub.rebootDetected:
 
-	return err
+	case <-time.After(time.Second * 4):
```

Comment:
Its been a while but I thought we'd found that time.After can be problematic and to use a context with timeout:
```
d := time.Now().Add(shortDuration)
	ctx, cancel := context.WithDeadline(context.Background(), d)

	// Even though ctx will be expired, it is good practice to call its
	// cancellation function in any case. Failure to do so may keep the
	// context and its parent alive longer than necessary.
	defer cancel()

	select {
	case <-time.After(1 * time.Second):
		fmt.Println("overslept")
	case <-ctx.Done():
		fmt.Println(ctx.Err())
	}
```

Reply:
time.After() can cause memory leaks by not being Garbaged collected...

Reply:
Confused - we have time.After in the original code, thats why I suggest case <-ctx.Done():